src/main/java/uk/gov/hmcts/reform/userprofileapi/util/IdamStatusResolver.java

```diff
@@ -43,11 +51,41 @@ public static String resolveStatusAndReturnMessage(HttpStatus httpStatus) {
                 return USER_EXISTS;
             case NO_CONTENT:
                 return NO_CONTENT;
+            case PRECONDITION_FAILED:
+                return PRECONDITION_FAILED;
             default:
                 return UNKNOWN;
         }
     }
 
+    public static String resolveStatusAndReturnMessage(ResponseEntity<Object> responseEntity) {
+        String errorMessage = null;
+        if (nonNull(responseEntity)) {
+            Object responseBody = responseEntity.getBody();
+            if (nonNull(responseBody) && responseBody instanceof IdamErrorResponse) {
+                errorMessage = getErrorMessageFromSidamResponse(responseBody);
```

Comment:
try to add inner if loop in the top  if possible ..i am not finding much difference to add it in the inner if

Reply:
condition is right , if responseEntity is null then defaulting responseEntity with INTER_SERVER_ERROR else if responseEntity is without body then take status code from it to resolve error message.